client/state/posts/utils.js

```diff
@@ -91,3 +98,60 @@ export function mergeIgnoringArrays( object, ...sources ) {
 		}
 	} );
 }
+
+/**
+ * Given a post object, returns a normalized post object
+ *
+ * @param  {Ojbect} post Raw edited post object
+ * @return {Object}      Normalized post object
+ */
+export function normalizePost( post ) {
+	if ( ! post ) {
+		return null;
+	}
+
+	return normalizeFlow( post );
+}
+
+/**
+ * Takes existing term post edits and updates the `terms_by_id` attribute
+ *
+ * @param  {Object}    post  object of post edits
+ * @return {Object}          normalized post edits
+ */
+export function getTermIdsFromEdits( post ) {
+	if ( ! post || ! post.terms ) {
+		return post;
+	}
+	const normalizedPost = cloneDeep( post );
```

Comment:
I think we could save on some performance by avoiding the `cloneDeep` here and using assign below, advantage of which being that only a shallow copy is created and we can return the original post (bypass the assign) when taxonomies is empty.

```js
return {
	...post,
	terms_by_id: mapValues( /* ... */ )
};
```